### PR TITLE
Implemented models and WeatherAPIService

### PR DESCRIPTION
**Pull request by deffsin**

### CODE REVIEW

File: WeatherApp/Data/Models/Weather.swift [hunk 1]
```diff
@@ -0,0 +1,75 @@
+//
+//  Weather.swift
+//  WeatherApp
+//
+//  Created by Denis Sinitsa on 29.08.2023.
+//
+
+import Foundation
+
+struct WeatherResponse: Decodable {
```
Comment (Hassanq911) [inline WeatherApp/Data/Models/Weather.swift:10]: We can name it as `WeatherResponseModel` so its more understandable

File: WeatherApp/Data/Models/Weather.swift [hunk 2]
```diff
@@ -0,0 +1,75 @@
...
+            description: self.weather.first?.description ?? "",
+            temperature: self.main.temp,
+            sunrise: self.sys.sunrise,
+            sunset: self.sys.sunset
+        )
+    }
+}
+
+
+struct WeatherMain: Decodable {
```
Comment (Hassanq911) [inline WeatherApp/Data/Models/Weather.swift:36]: Try to add the suffix `Model` at the end indicating its purpose when using else where in code

File: WeatherApp/Data/Models/Weather.swift [hunk 3]
```diff
@@ -0,0 +1,75 @@
...
+}
+
+struct WeatherIcon: Decodable {
+    let main: String
+    let description: String
+    let icon: String
+}
+
+
+struct Weather: Decodable {
```
Comment (Hassanq911) [inline WeatherApp/Data/Models/Weather.swift:68]: Whats the use for this struct. If its a `dto` for view then make a separate file for this

File: WeatherApp/Data/Models/Weather.swift [hunk 4]
```diff
@@ -0,0 +1,75 @@
...
+    let sys: Sys
+
+    private enum CodingKeys: String, CodingKey {
+        case city = "name"
+        case main
+        case weather
+        case sys
+    }
+
+    func toWeather() -> Weather {
```
Comment (Hassanq911) [inline WeatherApp/Data/Models/Weather.swift:23]: i would suggest making a separate extension for this in a file name `WeatherResponseModel+Mapping`

File: WeatherApp/Services/Network/WeatherAPIService.swift [hunk 5]
```diff
@@ -0,0 +1,28 @@
...
+//  Created by Denis Sinitsa on 29.08.2023.
+//
+
+import Foundation
+
+class WeatherAPIService {
+    private let networkService = NetworkService()
+    
+    func fetchWeather(forCity city: String, completion: @escaping ((Result<Weather, NetworkError>) -> Void)) {
+        guard let weatherURL = Constants.Urls.weatherURL(forCity: city) else {
```
Comment (Hassanq911) [inline WeatherApp/Services/Network/WeatherAPIService.swift:14]: Where is the Constants.URL here ?? Is the file already included in the project ?

**Review by Hassanq911 — CHANGES_REQUESTED** [review]

File: WeatherApp/Data/Models/Weather.swift (hunk 1 above)
Comment (deffsin) [reply WeatherApp/Data/Models/Weather.swift:10]: ✅

File: WeatherApp/Services/Network/WeatherAPIService.swift (hunk 5 above)
Comment (deffsin) [reply WeatherApp/Services/Network/WeatherAPIService.swift:14]: I forgot to add it.

**Review by Hassanq911 — CHANGES_REQUESTED** [review]

**Review by Hassanq911 — APPROVED** [review]

**Pull request merged**